### PR TITLE
Revert "Revert "[Build] Make cmark build a build-script product (#371…

### PR DESCRIPTION
**Pull request by drexin**
…02)""

This should fix the cmark build to respect cross compilation targets.

### CODE REVIEW

**Comment by drexin** [comment]
@swift-ci please build toolchain

**Comment by drexin** [comment]
@swift-ci please build toolchain

**Comment by drexin** [comment]
@swift-ci please build toolchain

**Comment by drexin** [comment]
@swift-ci please build toolchain

**Comment by drexin** [comment]
@swift-ci build toolchain macos

**Comment by drexin** [comment]
@swift-ci build toolchain

File: utils/swift_build_support/swift_build_support/products/earlyswiftdriver.py [hunk 1]
```diff
@@ -41,6 +41,9 @@ def is_before_build_script_impl_product(cls):
         return True
 
     def should_build(self, host_target):
+        if self.is_cross_compile_target(host_target):
```
Comment (gottesmm) [inline utils/swift_build_support/swift_build_support/products/earlyswiftdriver.py:44]: In a follow on commit, can you add a comment here? Also we should ping @artemcm and @nkcsgexi so they know about this.

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci build toolchain macos

**Comment by drexin** [comment]
@swift-ci build toolchain macos

**Review by gottesmm — APPROVED** [review]

**Pull request merged**